app/parsers/bulkrax/bagit_parser.rb

```diff
@@ -100,8 +100,17 @@ def works_total
       total
     end
 
+    # TODO: look into if CsvParser#total needs this same logic
```

Comment:
```suggestion
```

Reply:
this is already the exact logic in the csv parser.

Reply:
This was copied over but had to be changed because if `limit` was 0, then `@total` would come out as 0, if `limit` was nil then `@total` would be whatever `current_record_ids.count` is.  The user chooses the limit, but it is indicated that 0 or blank _should_ do the same thing.  It is seemingly not, at least in the `BagitParser`.

Reply:
mmm. I actually wouldn't expect a limit of 0 and a "missing' limit to do the same. to me "0" does not equal "none". 

I would expect a limit to be enforced only if a value were placed in the "limit" field. even if that value were 0. in which case my logic is in line with the logic in the csv parser. 

perhaps we should get another opinion?

Reply:
Ah gotcha, I was going off of what the UI says, but I admit I haven't explicitly tested what if I put a 0 in there.
![image](https://user-images.githubusercontent.com/19597776/174171003-f11f747b-257a-4a88-a2a9-46fd6247e5b9.png)

I can just take it out for now until further testing.

Reply:
you know what.....idk if I ever looked at that message. 😅  that's not at all how I'd expect it to work. if it is then you're implementation is what should happen.

Reply:
@kirkkwang I approved this pr...but we should still check with others what the intention here should be.

Reply:
Thanks @alishaevn and agreed!